src/main/java/Joseph/Tasks/TaskList.java

```diff
@@ -0,0 +1,165 @@
+package Joseph.Tasks;
+
+import Joseph.*;
+import Joseph.Exceptions.InsufficientDetailsException;
+
+import java.util.ArrayList;
+
+/**
+ * Contains the task list as well as operations for adding. deleting, marking and unmarking tasks.
+ * Handles parsed commands.
+ */
+public class TaskList {
+
+    private ArrayList<Task> tasks;
+
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    public TaskList(ArrayList<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    public ArrayList<Task> getTasks() {
+        return tasks;
+    }
+
+    /**
+     * Adds a task to the task list.
+     * @param task The task to be added
+     */
+    public void addTask(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Deletes a task from the task list.
+     * @param index The index of the task to be deleted. User input is 1-indexed.
+     * @return The deleted task.
+     * @throws IndexOutOfBoundsException If no task exists at the specified index.
+     */
+    public Task deleteTask(int index) throws IndexOutOfBoundsException {
+        return tasks.remove(index - 1);
+    }
+
+    /**
+     * Retrieves a task from the task list.
+     * @param index The index of the task to be retrieved. User input is 1-indexed.
+     * @return The task at the specified index.
+     * @throws IndexOutOfBoundsException If no task exists at the specified index.
+     */
+    public Task getTask(int index) throws IndexOutOfBoundsException {
+        return tasks.get(index - 1);
+    }
+
+    /**
+     * Marks a task as done.
+     * @param index The index of the task to be marked. User input is 1-indexed.
+     * @throws IndexOutOfBoundsException If no task exists at the specified index.
+     */
+    public void markTask(int index) throws IndexOutOfBoundsException {
+        tasks.get(index - 1).setDone();
+    }
+    /**
+     *
+     * Unmarks a task as not done.
+     * @param index The index of the task to be unmarked. User input is 1-indexed.
+     * @throws IndexOutOfBoundsException If no task exists at the specified index.
+     */
+    public void unmarkTask(int index) throws IndexOutOfBoundsException {
+        tasks.get(index - 1).unDone();
+    }
+
+    public int size() {
```

Comment:
Names representing methods should be verbs according to the java coding standard. Perhaps you could change it to getSize? 😅